File: DP_Tokenizer/DP_tokenizer/Token.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DP_Tokenizer
{
    public class Token
    {
        public int LineNumber { get; private set; }
        public int LinePosition { get; private set; }
        public int Level { get; private set; }
        public string Value { get; private set; }
        public TokenType Type { get; private set; }
        public String PartnerValue { get { return (Partner != null) ? Partner.Value : ""; } set { Partner.Value = value; } }
        public Token Partner { get; set; }

        public Token(int lineNumber, int linePosition, int level, string value, TokenType type, Token partner)
        {
            LineNumber      = lineNumber;
            LinePosition    = linePosition;
            Level           = level;
            Value           = value;
            Type            = type;
            Partner         = partner;
        }
    }
}

File: C++Tokenizer/C++Tokenizer/Token.cpp
//
//  Token.cpp
//  tokenizer
//
//  Created by Alex Nagelkerke on 19-09-14.
//  Copyright (c) 2014 Alex Nagelkerke. All rights reserved.
//

#include <stdio.h>
#include "Token.h"

Token::Token(int lineNumber, int linePosition, int level, std::string value, TokenType tokenType, Token *partner)
{
	LineNumber = lineNumber;
	LinePosition = linePosition;
	Level = level;
	Value = value;
	Type = tokenType;
	Partner = partner;
}

Token::Token()
{
	LineNumber = 0;
	LinePosition = 0;
	Level = 0;
	Value = "";
}